qctrlopencontrols/driven_controls/predefined.py

```diff
@@ -266,13 +289,12 @@ def _new_wimperis_1_control(
     qctrlopencontrols.DrivenControl
         The driven control.
     """
-    (maximum_rabi_rate, rabi_rotation, azimuthal_angle) = _predefined_common_attributes(
-        maximum_rabi_rate, rabi_rotation, azimuthal_angle
+    (maximum_rabi_rate, azimuthal_angle, rabi_rotation) = _predefined_common_attributes(
+        maximum_rabi_rate, azimuthal_angle, rabi_rotation
     )
-
-    phi_p = _get_transformed_rabi_rotation_wimperis(rabi_rotation)
-
-    rabi_rotations = [rabi_rotation, np.pi, 2 * np.pi, np.pi]
+    if rabi_rotation is not None:
```

Comment:
Any idea why the type checker isn't picking up that line 302 could fail if `rabi_rotation` is `None`, since `phi_p` won't get defined?

Reply:
Not sure. I was quite surprised to see such an obvious error just passes mypy checking.

Reply:
https://github.com/python/mypy/issues/2400

Crazy stuff.

Reply:
indeed....